importer/wdi_importer.py

```diff
@@ -1,63 +1,922 @@
-                sources_dict[cell.value] = definition
-
-    column_number = 0
-
-print(json.dumps(global_cat))
+from openpyxl import load_workbook
+from grapher_admin.models import Entity, DatasetSubcategory, DatasetCategory, Dataset, Source, Variable, VariableType, DataValue
+from importer.models import ImportHistory, AdditionalCountryInfo
+from country_name_tool.models import CountryName
+from django.db import connection
+from django.utils import timezone
+from django.urls import reverse
+
+
+# we will use the file checksum to check if the downloaded file has changed since we last saw it
+def file_checksum(filename, blocksize=2**20):
+    m = hashlib.md5()
+    with open(filename, "rb") as f:
+        while True:
+            buffer = f.read(blocksize)
+            if not buffer:
+                break
+            m.update(buffer)
+    return m.hexdigest()
+
+
+source_template = '<table>' \
+                    '<tr>' \
+                        '<td>Variable category</td>' \
+                        '<td>%s</td>' \
+                    '</tr>' \
+                    '<tr>' \
+                        '<td>Unit of measure</td>' \
+                        '<td>%s</td>' \
+                    '</tr>' \
+                    '<tr>' \
+                        '<td>Variable code in original source</td>' \
+                        '<td>%s</td>' \
+                    '</tr>' \
+                    '<tr>' \
+                        '<td>Data published by</td>' \
+                        '<td>%s</td>' \
+                    '</tr>' \
+                    '<tr>' \
+                        '<td>Data publisher\'s source</td>' \
+                        '<td>%s</td>' \
+                    '</tr>' \
+                    '<tr>' \
+                        '<td>Link</td>' \
+                        '<td>http://data.worldbank.org/data-catalog/world-development-indicators</td>' \
+                    '</tr>' \
+                    '<tr>' \
+                        '<td>Retrieved</td>' \
+                        '<td>' + timezone.now().strftime("%d-%B-%y") +'</td>' \
+                    '</tr>' \
+                  '</table>' \
+                  '<div class="datasource-additional">' \
+                    '<p>' \
+                    '<b><i style="text-decoration: underline;">Additional information as provided by the source</i></b><br>' \
+                    '<i style="text-decoration: underline;">Definitions and characteristics of countries and other territories:</i> <a href="' + reverse("servewdicountryinfo") + '">WDI_Country_info.xlsx</a><br>' \
+                    '%s' \
+                    '%s' \
+                    '%s' \
+                    '%s' \
+                    '%s' \
+                    '%s' \
+                    '</p>' \
+                  '</div>'
+
+
+wdi_zip_file_url = 'http://databank.worldbank.org/data/download/WDI_excel.zip'
+wdi_downloads_save_location = os.path.dirname(os.path.realpath(__file__)) + '/wdi_downloads/'
```

Comment:
Note that the grapher repo gets overwritten on deployment, so these directories will be deleted unless they're symlinked in by deploy.sh.

I'd suggest we keep any persistent output data in the one location (to avoid symlinks all over the place)-- probably BASE_DIR + '/data' or similar